src/execution_impl.rs

```diff
@@ -13,88 +16,119 @@ use wasmer::{
 use wasmer_as::{Read as ASRead, StringPtr, Write as ASWrite};
 use wasmer_middlewares::Metering;
 
-/// `Print` ABI called by the webassembly VM
-fn print(env: &Env, arg: i32) {
-    let str_ptr = StringPtr::new(arg as u32);
-    println!(
-        "{}",
-        str_ptr
-            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
-            .unwrap()
-    );
+/// Utility function to call a WASM module (compiled from AssemblyScript) using
+/// high-level Rust types (by de/serializing a String).
+///
+/// The intend is to be able to expose any kind of function as an ABI:
+///
+/// ```ignore
+/// fn get_status() -> Result<NodeStatus> {
+///     Ok(typed_call(env, "0xAddressOfSCInLedger", "get_status", ())?)
+/// }
+/// ```
+pub fn typed_call<T: Serialize, R: DeserializeOwned>(
+    env: &Env,
+    address: &Address,
+    function: &str,
+    _args: T, // TODO: @adrien-zinger how do we pass function args in AssemblyScript?
+) -> Result<R> {
+    let value = call_module(env, address, function);
+    Ok(serde_json::from_str::<R>(&value.ret)?)
 }
 
 /// `Call` ABI called by the webassembly VM
+///
+/// Call an exported function in a WASM module at a given address
+///
 /// It take in argument the environment defined in env.rs
 /// this environment is automatically filled by the wasmer library
 /// And two pointers of string. (look at the readme in the wasm folder)
```

Comment:
Can you add the link here ? Is there a way to make in shinny in the generated doc ?

Reply:
I add an `#[doc = include_str!("../wasm/README.md")]` just beside! Should I move it here?